spring-cloud-open-service-broker-autoconfigure/src/main/java/org/springframework/cloud/servicebroker/autoconfigure/web/Plan.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.servicebroker.autoconfigure.web;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.hibernate.validator.constraints.NotEmpty;
+
+/**
+ * A service plan available for a ServiceDefinition
+ *
+ * @author sgreenberg@pivotal.io
+ * @author Scott Frederick
+ * @author Roy Clarkson
+ */
+class Plan {
+
+	/**
+	 * An identifier used to correlate this plan in future requests to the catalog. This must be unique within
+	 * the platform. Using a GUID is recommended.
+	 */
+	@NotEmpty
```

Comment:
These validation annotations are honored by Jackson when unmarshalling objects from JSON. Do they have any value in Spring configuration properties?

Reply:
Yeah, it's a similar behavior for configuration properties. It'll throw an exception if the configuration doesn't pass validation. https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-external-config-validation.